flink-java/src/main/java/org/apache/flink/api/java/functions/SelectByMinFunction.java

```diff
@@ -48,7 +49,7 @@ public SelectByMinFunction(TupleTypeInfo<T> type, int... fields) {
 			}
 
 			// Check whether type is comparable
-			if (!type.getTypeAt(field).isKeyType()) {
+			if (!((TupleTypeInfo)type).getTypeAt(field).isKeyType()) {
```

Comment:
Scala tuple types are not of type `TupleTypeInfo` but instead `TupleTypeInfoBase`.